src/compiler/checker.ts

```diff
@@ -11708,6 +11710,29 @@ namespace ts {
             return annotationIncludesUndefined ? getTypeWithFacts(declaredType, TypeFacts.NEUndefined) : declaredType;
         }
 
+        function isApparentTypePosition(node: Node) {
+            const parent = node.parent;
+            return parent.kind === SyntaxKind.PropertyAccessExpression ||

```

Comment:
No need to. This function is only ever called on the left hand side of a property access expression (the right hand side is just an identifier and not actually an expression).